atlasdb-client/src/main/java/com/palantir/atlasdb/table/description/TableDefinition.java

```diff
@@ -62,34 +62,86 @@ public void constraints() {
         state = State.DEFINING_CONSTRAINTS;
     }
 
+    /**
+     * Indicates that the name of the table being defined is safe for logging.
+     */
+    public void tableNameIsSafeLoggable() {
```

Comment:
think we should add one more option that is a superset, i.e., `allSafeForLoggingByDefault()`?

this is probably what most users will do, so might as well avoid needing two lines. although in this case i think we need a `tableNameLogSafety()` method to override that default